### PR TITLE
[BD-6] Compliance with OEP-18

### PR DESCRIPTION
**Pull request by ericfab179**
Update setup.py install_requirements definition.
Update travis file using new requirements.
Update tox file using new requirements.
create pip_tools, base, doc, test, travis and tox requirements files.
Create Makefile and add upgrade command.
Include requirements files generated using upgrade command.
Update openedx.yml OEPs list.

Add python 3.8 to test.

**Reviewers**
- [x] @morenol
- [ ] @awais786 
- [ ] @jmbowman

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ericfab179! I've created [OSPR-4454](https://openedx.atlassian.net/browse/OSPR-4454) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: .travis.yml [hunk 1]
```diff
@@ -3,30 +3,17 @@ sudo: false
 language: python
 
 python:
-  - 2.7
   - 3.5
-  - 3.6
+  - 3.8
 
 env:
```
Comment (morenol) [inline .travis.yml:9]: Add the quality tests

File: tox.ini [hunk 2]
```diff
@@ -7,11 +7,7 @@ setenv =
     PYTHONPATH = {toxinidir}
 
 deps =
-    django111: Django>=1.11,<2.0
-    django20: Django>=2.0,<2.1
-    django21: Django>=2.1,<2.2
-    django22: Django>=2.2,<2.3
```
Comment (morenol) [inline tox.ini:13]: Do not remove this one and add the django30 option. You should remove from test.txt the line with the django version in order to control which version is used, See https://github.com/edx/edx-enterprise/blob/6f9b8535cd68ee092eb389d53f69b6e05aa2e3a2/Makefile#L92 for reference

**Comment by edx-webhook** [comment]
@ericfab179 Even though your pull request wasn’t merged, please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-django-sites-extensions&entry.867055334=https://github.com/edx/edx-django-sites-extensions/pull/25&entry.1484655318=https://github.com/ericfab179&entry.752974735=2020-05-03+23:59&entry.1917517419=2020-05-04+22:33&entry.2133058324=No) so we can improve your experience in the future.

**Comment by morenol** [comment]
@ericfab179 please squash this. After that I think that it is ready

**Pull request merged**